### ISSUE TITLE
ffi-1.9.10/lib/ffi/library.rb:261:in `attach_function': Function 'unorm2_getInstance_4_2' not found in [/usr/lib64/libicui18n.so.42, /usr/lib64/libicutu.so.42] (FFI::NotFoundError)

### ISSUE THREAD

**Issue by JoeJennings** [jarib/ffi-icu#26]
I’m noticing the following error with ffi-icu 0.1.8 in a gem when running rspec tests, here is an example failure:

```
rake aborted!
/usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/ffi-1.9.10/lib/ffi/library.rb:261:in `attach_function': Function 'unorm2_getInstance_4_2' not found in [/usr/lib64/libicui18n.so.42, /usr/lib64/libicutu.so.42] (FFI::NotFoundError)
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/ffi-icu-0.1.8/lib/ffi-icu/lib.rb:345:in `<module:Lib>'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/ffi-icu-0.1.8/lib/ffi-icu/lib.rb:8:in `<module:ICU>'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/ffi-icu-0.1.8/lib/ffi-icu/lib.rb:1:in `<top (required)>'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/activesupport-4.2.4/lib/active_support/dependencies.rb:274:in `require'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/activesupport-4.2.4/lib/active_support/dependencies.rb:274:in `block in require'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/activesupport-4.2.4/lib/active_support/dependencies.rb:240:in `load_dependency'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/activesupport-4.2.4/lib/active_support/dependencies.rb:274:in `require'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/ffi-icu-0.1.8/lib/ffi-icu.rb:33:in `<top (required)>'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/activesupport-4.2.4/lib/active_support/dependencies.rb:274:in `require'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/activesupport-4.2.4/lib/active_support/dependencies.rb:274:in `block in require'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/activesupport-4.2.4/lib/active_support/dependencies.rb:240:in `load_dependency'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/activesupport-4.2.4/lib/active_support/dependencies.rb:274:in `require'
	from /home/ion/jenkins/workspace/test_engine/lib/test_engine.rb:7:in `<top (required)>'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/activesupport-4.2.4/lib/active_support/dependencies.rb:274:in `require'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/activesupport-4.2.4/lib/active_support/dependencies.rb:274:in `block in require'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/activesupport-4.2.4/lib/active_support/dependencies.rb:240:in `load_dependency'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/activesupport-4.2.4/lib/active_support/dependencies.rb:274:in `require'
	from /home/ion/jenkins/workspace/test_engine/spec/rails_helper.rb:10:in `<top (required)>'
	from /home/ion/jenkins/workspace/test_engine/spec/controllers/test/test_controller_spec.rb:1:in `require'
	from /home/ion/jenkins/workspace/test_engine/spec/controllers/test/test_controller_spec.rb:1:in `<top (required)>'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/rspec-core-3.3.2/lib/rspec/core/configuration.rb:1327:in `load'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/rspec-core-3.3.2/lib/rspec/core/configuration.rb:1327:in `block in load_spec_files'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/rspec-core-3.3.2/lib/rspec/core/configuration.rb:1325:in `each'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/rspec-core-3.3.2/lib/rspec/core/configuration.rb:1325:in `load_spec_files'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/rspec-core-3.3.2/lib/rspec/core/runner.rb:102:in `setup'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/rspec-core-3.3.2/lib/rspec/core/runner.rb:88:in `run'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/rspec-core-3.3.2/lib/rspec/core/runner.rb:73:in `run'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/rspec-core-3.3.2/lib/rspec/core/runner.rb:41:in `invoke'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/gems/rspec-core-3.3.2/exe/rspec:4:in `<top (required)>'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/bin/rspec:23:in `load'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/bin/rspec:23:in `<main>'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/bin/ruby_executable_hooks:15:in `eval'
	from /usr/local/rvm/gems/ruby-2.1.3@test_engine/bin/ruby_executable_hooks:15:in `<main>'
Coverage report generated for RSpec to /home/ion/jenkins/workspace/test_engine/coverage. 624 / 810 LOC (77.04%) covered.
```

This error corresponds to using ffi-icu 0.1.8 that appears to have became available in RubyGems[1] on Oct 1, not sure why date in RubyGems says Aug 23, 2010, however I also I did not see a tag version on the gitHub repo [2] for 0.1.8.

[1] https://rubygems.org/gems/ffi-icu/versions/0.1.8
[2] https://github.com/jarib/ffi-icu/releases

My test project is using these dependencies:
rspec-rails 3.3.3
rake 0.9.6
ffi 1.9.10

**Comment by jarib** [comment]
@fantasticfears Any idea about this?

**Comment by fantasticfears** [comment]
The version of your ICU library is 4.2. However, this new sets of API is available since 4.4.
Given the newest ICU library version is 55, I didn't protect this by version check which cause the failure spec.

@jarib What's the ICU version `ffi-icu` required by the way? I can fix it by a version check.

**Comment by jarib** [comment]
@JoeJennings Please try version 0.1.9 and report back.

@fantasticfears I'm not really sure what version is required - we haven't set a hard requirement before.
